NEWAPI.md

```diff
@@ -32,124 +31,222 @@ know how to use `git` appropriately.
+are not automatically populated:
 
-The `protoPath` property is the directory within the `googleapis` repo, e.g.
-`"google/firestore/v1"`.
+- Tags
+- The product URL (for documentation)
 
-The above assumes you're happy to create an initial beta release for
-the generated code immediately. If you want to avoid creating a
-release right now, use a version of `1.0.0-beta00`.
+These should be edited into apis.json by hand at some point before
+the first release. Note that the tooling sets the version to
+"1.0.0-beta00" by default, which is effectively "the version before
+the first beta release".
 
-If your project uses the IAM or long-running operations APIs, you'll need to add dependencies for those, e.g.
+Once you're happy with the change, commit it with an explanatory
+message, e.g.
 
-```json
-"dependencies": {
-  "Google.LongRunning": "2.0.0",
-  "Google.Cloud.Iam.V1": "2.0.0"
-}
+```sh
+git commit -a -m "Add Google.Cloud.Dialogflow.Cx.V3 to the API catalog"
 ```
 
-Look at other APIs for example values.
+Step 5: Generate and build the source code and projects
+-------------------------------------------------------
 
-Step 4: Run generateapis.sh
----------------------------
+There are two generation steps here: generating the C# files, and
+then generating the project files to go along with them.
 
-You can either run this by specifying the IDs of the packages you
-want to generate, or run it without specifying any arguments at all,
-in which case all APIs will be generated.
+First, run `generateapis.sh`, specifying the ID of the package:
 
-This will clone both the `googleapis` and `gapic-generator-csharp` repos as
-subdirectories, or pull them if they already exist.
+```sh
+./generateapis.sh Google.Cloud.Dialogflow.Cx.V3
+```
 
-Step 5: Generate project files
-----------------------
+If all goes well, this should just write (after a few seconds):
 
-Run `./generateprojects.sh`. This should create:
+```text
+Generating Google.Cloud.Dialogflow.Cx.V3
+```
+
+That step generates just the C# source code. Next, generate the
+project files:
+
+```sh
+./generateprojects.sh
+```
+
+(Note that you don't need to specify the package name here - it
+regenerates *all* the projects - as well as a few other files, if
+necessary.)
 
-- A solution file
-- Project files for the production project and the snippets
-- A stub documentation file
+Use `git status` to check that a new directory has been created
+(`apis/Google.Cloud.Dialogflow.Cx.V3` in this example).
 
-Step 6: Build the code
---------------------------
+Now we can build the package and run the unit tests:
 
-Run `./build.sh Your.ApiName.Here` to check it builds and the
-generated unit tests pass.
+```sh
+./build.sh Google.Cloud.Dialogflow.Cx.V3
+```
+
+If anything fails here, it's likely to be either an API
+configuration problem, or a generator problem. Either way, raise a
+GitHub issue - unless you're feeling particularly patient, it's
```

Comment:
nit: inpatient?

Reply:
Nope, this is deliberate. *Unless* you're feeling particularly patient (as in, you're willing to do a lot of debugging work yourself)

Reply:
Ha, I thought it was meant to say "Unless you're feeling particularly impatient" (as in, you don't want to wait for someone in the team to look at it), haha.